sklearn/svm/src/liblinear/linear.cpp

```diff
@@ -70,6 +78,33 @@ static void info(const char *fmt,...)
 static void info(const char *fmt,...) {}
 #endif
 
+// New function to ensure the same behaviour for random number generation on windows and linux
+int myrand() {
+#ifdef _WIN32
+	// In MS Visual Studio (2012) RAND_MAX = 0x7FFF (15bit) = 32767
+	// In Linux GCC (4.6) 32bits RAND_MAX = 0x7FFFFFFF (31bit) = 2147483647
+	// In Linux GCC (4.6) 64bits RAND_MAX = 0x7FFFFFFFFFFFFFFF (63 bits) = 9223372036854775807
+	// so in MS Visual Studio we need to call rand() several times to always ensure the same random number range than in Linux GCC
+	if (std::numeric_limits<int>::max() == 0x7FFFFFFF) {
```

Comment:
Can this be determined at compile time?

Reply:
I'm *really* not a C++ expert but from what I've read, apparently not: the boost library even proposes something [to fix this issue](https://www.boost.org/doc/libs/1_34_0/libs/integer/integer_traits.html).

Reply:
Rather than using `# ifdef _WIN32` and then do this dynamic check at runtime, I will try to use `#if RAND_MAX != INT_MAX && INT_MAX == 0x7FFFFFFF` for the first case and `#if RAND_MAX != INT_MAX && INT_MAX == 0x7FFFFFFFFFFFFFFF` for the second.

Reply:
Solution seems good now.